addons/website/static/src/js/backend/view_hierarchy/view_hierarchy.js

```diff
@@ -0,0 +1,249 @@
+/** @odoo-module **/
+
+import { getDefaultConfig } from "@web/views/view";
+import { HierarchyNavbar } from "./hierarchy_navbar";
+import { Layout } from "@web/search/layout";
+import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
+
+const { Component, onWillStart, useEffect, useState, useSubEnv } = owl;
+
+export class ViewHierarchy extends Component {
+    setup() {
+        this.action = useService("action");
+        this.hideGenericViewByWebsite = {};
+        this.orm = useService("orm");
+        this.router = useService("router");
+        this.state = useState({ showInactive: false, searchedView: {}, viewTree: {} });
+        this.viewId = this.props.action.context.active_id;
+        this.websites = useState({ names: new Set(["All Websites"]), selected: "All Websites" });
+
+        onWillStart(async () => {
+            const { sibling_views, hierarchy } = await this.orm.call(
+                "ir.ui.view",
+                "get_view_hierarchy",
+                [this.viewId],
+                {}
+            );
+            this.state.viewTree = hierarchy;
+            this.siblingViews = sibling_views;
+
+            this.loadWebsiteNames();
+            this.loadHideGenericViewByWebsite();
```

Comment:
"load" kind of indicates that we fetch something from the server, even though these functions look synchronous, maybe bad naming?